enhancements/monitoring/alert-overrides.md

```diff
@@ -0,0 +1,576 @@
+  choosing between, e.g. from `Warning` to `Critical`.
+- As an OpenShift user, I want to add labels to alerts to aide in routing and
+  triage.
+- As an OpenShift user, I want a supported way to add custom alerts using
+  metrics from components in the `openshift-*` namespaces.
+- As an OpenShift developer, I want the alerting rules shipped by my component
+  to be evaluated and forwarded to telemetry even when overridden by a user.
+
+### API Extensions
+
+We will be adding two new CRDs for alerting rules and relabeling:
+
+- `alertingrule.monitoring.openshift.io/v1alpha1`
+- `alertrelabelconfig.monitoring.openshift.io/v1alpha1`
+
+```go
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+
+// NOTE: The AlertingRule type is a direct copy of the upstream PrometheusRule
+// type from prometheus-operator.  The only difference at the moment is that we
+// don't allow recording rules in OpenShift.  All rules must be alerting rules,
+// but outside of that restriction, each AlertingRule will result in a 1:1 alike
+// PrometheusRule object being created.
+//
+// See the upstream docs here:
+// - https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md
+
+// AlertingRule represents a set of user-defined Prometheus rule groups containing
+// alerting rules -- recording rules are not allowed.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +k8s:openapi-gen=true
+type AlertingRule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec describes the desired state of this AlertingRule object.
+	Spec AlertingRuleSpec `json:"spec"`
+
+	// status describes the current state of this AlertOverrides object.
+	//
+	// +optional
+	Status AlertingRuleStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AlertingRuleList is a list of AlertingRule objects.
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+// +k8s:openapi-gen=true
+type AlertingRuleList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// items is a list of AlertingRule objects.
+	Items []AlertingRule `json:"items"`
+}
+
+// AlertingRuleSpec is the desired state of an AlertingRule resource.
+//
+// +k8s:openapi-gen=true
+type AlertingRuleSpec struct {
+	// groups is a list of grouped alerting rules.
```

Comment:
Might be worth adding a couple of helpers for server side apply, this depends on how you are expecting this list to be managed. Is it likely multiple owners might add entries?

The following would mean that multiple owners can configure their own items in this list without affecting the other entries
```
	// +listType=map
	// +listMapKey=name
```
Else the list is considered atomic and any user updating the list overwrites the whole list